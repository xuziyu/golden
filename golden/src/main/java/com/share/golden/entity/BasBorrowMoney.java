package com.share.golden.entity;

public class BasBorrowMoney {
    private String borrowId;

    private String borrowSerial;

    private String userId;

    private String borrowCompany;

    private String borrowMoney;

    private String borrowRate;

    private String timeLimit;

    private String borrowLimit;

    private String borrowState;

    private String borrowRecommend;

    private String borrowType;

    private String borrowRemark1;

    private String borrowRemark2;

    public String getBorrowId() {
        return borrowId;
    }

    public void setBorrowId(String borrowId) {
        this.borrowId = borrowId == null ? null : borrowId.trim();
    }

    public String getBorrowSerial() {
        return borrowSerial;
    }

    public void setBorrowSerial(String borrowSerial) {
        this.borrowSerial = borrowSerial == null ? null : borrowSerial.trim();
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId == null ? null : userId.trim();
    }

    public String getBorrowCompany() {
        return borrowCompany;
    }

    public void setBorrowCompany(String borrowCompany) {
        this.borrowCompany = borrowCompany == null ? null : borrowCompany.trim();
    }

    public String getBorrowMoney() {
        return borrowMoney;
    }

    public void setBorrowMoney(String borrowMoney) {
        this.borrowMoney = borrowMoney == null ? null : borrowMoney.trim();
    }

    public String getBorrowRate() {
        return borrowRate;
    }

    public void setBorrowRate(String borrowRate) {
        this.borrowRate = borrowRate == null ? null : borrowRate.trim();
    }

    public String getTimeLimit() {
        return timeLimit;
    }

    public void setTimeLimit(String timeLimit) {
        this.timeLimit = timeLimit == null ? null : timeLimit.trim();
    }

    public String getBorrowLimit() {
        return borrowLimit;
    }

    public void setBorrowLimit(String borrowLimit) {
        this.borrowLimit = borrowLimit == null ? null : borrowLimit.trim();
    }

    public String getBorrowState() {
        return borrowState;
    }

    public void setBorrowState(String borrowState) {
        this.borrowState = borrowState == null ? null : borrowState.trim();
    }

    public String getBorrowRecommend() {
        return borrowRecommend;
    }

    public void setBorrowRecommend(String borrowRecommend) {
        this.borrowRecommend = borrowRecommend == null ? null : borrowRecommend.trim();
    }

    public String getBorrowType() {
        return borrowType;
    }

    public void setBorrowType(String borrowType) {
        this.borrowType = borrowType == null ? null : borrowType.trim();
    }

    public String getBorrowRemark1() {
        return borrowRemark1;
    }

    public void setBorrowRemark1(String borrowRemark1) {
        this.borrowRemark1 = borrowRemark1 == null ? null : borrowRemark1.trim();
    }

    public String getBorrowRemark2() {
        return borrowRemark2;
    }

    public void setBorrowRemark2(String borrowRemark2) {
        this.borrowRemark2 = borrowRemark2 == null ? null : borrowRemark2.trim();
    }
}