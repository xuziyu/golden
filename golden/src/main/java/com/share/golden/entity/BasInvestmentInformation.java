package com.share.golden.entity;

import java.math.BigDecimal;
import java.util.Date;

public class BasInvestmentInformation {
    private String investorId;

    private String userId;

    private String tenderId;

    private BigDecimal investorMoney;

    private String investorStatus;

    private String investorStyle;

    private String brrowStatus;

    private String investorRest;

    private String profitModel;

    private BigDecimal profitMoney;

    private Date investorDate;

    private String replayDate;

    private String investorType;

    public String getInvestorId() {
        return investorId;
    }

    public void setInvestorId(String investorId) {
        this.investorId = investorId == null ? null : investorId.trim();
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId == null ? null : userId.trim();
    }

    public String getTenderId() {
        return tenderId;
    }

    public void setTenderId(String tenderId) {
        this.tenderId = tenderId == null ? null : tenderId.trim();
    }

    public BigDecimal getInvestorMoney() {
        return investorMoney;
    }

    public void setInvestorMoney(BigDecimal investorMoney) {
        this.investorMoney = investorMoney;
    }

    public String getInvestorStatus() {
        return investorStatus;
    }

    public void setInvestorStatus(String investorStatus) {
        this.investorStatus = investorStatus == null ? null : investorStatus.trim();
    }

    public String getInvestorStyle() {
        return investorStyle;
    }

    public void setInvestorStyle(String investorStyle) {
        this.investorStyle = investorStyle == null ? null : investorStyle.trim();
    }

    public String getBrrowStatus() {
        return brrowStatus;
    }

    public void setBrrowStatus(String brrowStatus) {
        this.brrowStatus = brrowStatus == null ? null : brrowStatus.trim();
    }

    public String getInvestorRest() {
        return investorRest;
    }

    public void setInvestorRest(String investorRest) {
        this.investorRest = investorRest == null ? null : investorRest.trim();
    }

    public String getProfitModel() {
        return profitModel;
    }

    public void setProfitModel(String profitModel) {
        this.profitModel = profitModel == null ? null : profitModel.trim();
    }

    public BigDecimal getProfitMoney() {
        return profitMoney;
    }

    public void setProfitMoney(BigDecimal profitMoney) {
        this.profitMoney = profitMoney;
    }

    public Date getInvestorDate() {
        return investorDate;
    }

    public void setInvestorDate(Date investorDate) {
        this.investorDate = investorDate;
    }

    public String getReplayDate() {
        return replayDate;
    }

    public void setReplayDate(String replayDate) {
        this.replayDate = replayDate == null ? null : replayDate.trim();
    }

    public String getInvestorType() {
        return investorType;
    }

    public void setInvestorType(String investorType) {
        this.investorType = investorType == null ? null : investorType.trim();
    }
}