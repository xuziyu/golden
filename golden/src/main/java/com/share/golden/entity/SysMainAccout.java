package com.share.golden.entity;

import java.math.BigDecimal;
import java.util.Date;

public class SysMainAccout {
    private String dealId;

    private String bankcardId;

    private String userId;

    private BigDecimal balance;

    private String dealType;

    private Date dealDate;

    public String getDealId() {
        return dealId;
    }

    public void setDealId(String dealId) {
        this.dealId = dealId == null ? null : dealId.trim();
    }

    public String getBankcardId() {
        return bankcardId;
    }

    public void setBankcardId(String bankcardId) {
        this.bankcardId = bankcardId == null ? null : bankcardId.trim();
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId == null ? null : userId.trim();
    }

    public BigDecimal getBalance() {
        return balance;
    }

    public void setBalance(BigDecimal balance) {
        this.balance = balance;
    }

    public String getDealType() {
        return dealType;
    }

    public void setDealType(String dealType) {
        this.dealType = dealType == null ? null : dealType.trim();
    }

    public Date getDealDate() {
        return dealDate;
    }

    public void setDealDate(Date dealDate) {
        this.dealDate = dealDate;
    }
}