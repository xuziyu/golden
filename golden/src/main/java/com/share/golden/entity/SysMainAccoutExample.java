package com.share.golden.entity;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class SysMainAccoutExample {
    protected String orderByClause;

    protected boolean distinct;

    protected List<Criteria> oredCriteria;

    public SysMainAccoutExample() {
        oredCriteria = new ArrayList<Criteria>();
    }

    public void setOrderByClause(String orderByClause) {
        this.orderByClause = orderByClause;
    }

    public String getOrderByClause() {
        return orderByClause;
    }

    public void setDistinct(boolean distinct) {
        this.distinct = distinct;
    }

    public boolean isDistinct() {
        return distinct;
    }

    public List<Criteria> getOredCriteria() {
        return oredCriteria;
    }

    public void or(Criteria criteria) {
        oredCriteria.add(criteria);
    }

    public Criteria or() {
        Criteria criteria = createCriteriaInternal();
        oredCriteria.add(criteria);
        return criteria;
    }

    public Criteria createCriteria() {
        Criteria criteria = createCriteriaInternal();
        if (oredCriteria.size() == 0) {
            oredCriteria.add(criteria);
        }
        return criteria;
    }

    protected Criteria createCriteriaInternal() {
        Criteria criteria = new Criteria();
        return criteria;
    }

    public void clear() {
        oredCriteria.clear();
        orderByClause = null;
        distinct = false;
    }

    protected abstract static class GeneratedCriteria {
        protected List<Criterion> criteria;

        protected GeneratedCriteria() {
            super();
            criteria = new ArrayList<Criterion>();
        }

        public boolean isValid() {
            return criteria.size() > 0;
        }

        public List<Criterion> getAllCriteria() {
            return criteria;
        }

        public List<Criterion> getCriteria() {
            return criteria;
        }

        protected void addCriterion(String condition) {
            if (condition == null) {
                throw new RuntimeException("Value for condition cannot be null");
            }
            criteria.add(new Criterion(condition));
        }

        protected void addCriterion(String condition, Object value, String property) {
            if (value == null) {
                throw new RuntimeException("Value for " + property + " cannot be null");
            }
            criteria.add(new Criterion(condition, value));
        }

        protected void addCriterion(String condition, Object value1, Object value2, String property) {
            if (value1 == null || value2 == null) {
                throw new RuntimeException("Between values for " + property + " cannot be null");
            }
            criteria.add(new Criterion(condition, value1, value2));
        }

        public Criteria andDealIdIsNull() {
            addCriterion("\"DEAL_ID\" is null");
            return (Criteria) this;
        }

        public Criteria andDealIdIsNotNull() {
            addCriterion("\"DEAL_ID\" is not null");
            return (Criteria) this;
        }

        public Criteria andDealIdEqualTo(String value) {
            addCriterion("\"DEAL_ID\" =", value, "dealId");
            return (Criteria) this;
        }

        public Criteria andDealIdNotEqualTo(String value) {
            addCriterion("\"DEAL_ID\" <>", value, "dealId");
            return (Criteria) this;
        }

        public Criteria andDealIdGreaterThan(String value) {
            addCriterion("\"DEAL_ID\" >", value, "dealId");
            return (Criteria) this;
        }

        public Criteria andDealIdGreaterThanOrEqualTo(String value) {
            addCriterion("\"DEAL_ID\" >=", value, "dealId");
            return (Criteria) this;
        }

        public Criteria andDealIdLessThan(String value) {
            addCriterion("\"DEAL_ID\" <", value, "dealId");
            return (Criteria) this;
        }

        public Criteria andDealIdLessThanOrEqualTo(String value) {
            addCriterion("\"DEAL_ID\" <=", value, "dealId");
            return (Criteria) this;
        }

        public Criteria andDealIdLike(String value) {
            addCriterion("\"DEAL_ID\" like", value, "dealId");
            return (Criteria) this;
        }

        public Criteria andDealIdNotLike(String value) {
            addCriterion("\"DEAL_ID\" not like", value, "dealId");
            return (Criteria) this;
        }

        public Criteria andDealIdIn(List<String> values) {
            addCriterion("\"DEAL_ID\" in", values, "dealId");
            return (Criteria) this;
        }

        public Criteria andDealIdNotIn(List<String> values) {
            addCriterion("\"DEAL_ID\" not in", values, "dealId");
            return (Criteria) this;
        }

        public Criteria andDealIdBetween(String value1, String value2) {
            addCriterion("\"DEAL_ID\" between", value1, value2, "dealId");
            return (Criteria) this;
        }

        public Criteria andDealIdNotBetween(String value1, String value2) {
            addCriterion("\"DEAL_ID\" not between", value1, value2, "dealId");
            return (Criteria) this;
        }

        public Criteria andBankcardIdIsNull() {
            addCriterion("\"BANKCARD_ID\" is null");
            return (Criteria) this;
        }

        public Criteria andBankcardIdIsNotNull() {
            addCriterion("\"BANKCARD_ID\" is not null");
            return (Criteria) this;
        }

        public Criteria andBankcardIdEqualTo(String value) {
            addCriterion("\"BANKCARD_ID\" =", value, "bankcardId");
            return (Criteria) this;
        }

        public Criteria andBankcardIdNotEqualTo(String value) {
            addCriterion("\"BANKCARD_ID\" <>", value, "bankcardId");
            return (Criteria) this;
        }

        public Criteria andBankcardIdGreaterThan(String value) {
            addCriterion("\"BANKCARD_ID\" >", value, "bankcardId");
            return (Criteria) this;
        }

        public Criteria andBankcardIdGreaterThanOrEqualTo(String value) {
            addCriterion("\"BANKCARD_ID\" >=", value, "bankcardId");
            return (Criteria) this;
        }

        public Criteria andBankcardIdLessThan(String value) {
            addCriterion("\"BANKCARD_ID\" <", value, "bankcardId");
            return (Criteria) this;
        }

        public Criteria andBankcardIdLessThanOrEqualTo(String value) {
            addCriterion("\"BANKCARD_ID\" <=", value, "bankcardId");
            return (Criteria) this;
        }

        public Criteria andBankcardIdLike(String value) {
            addCriterion("\"BANKCARD_ID\" like", value, "bankcardId");
            return (Criteria) this;
        }

        public Criteria andBankcardIdNotLike(String value) {
            addCriterion("\"BANKCARD_ID\" not like", value, "bankcardId");
            return (Criteria) this;
        }

        public Criteria andBankcardIdIn(List<String> values) {
            addCriterion("\"BANKCARD_ID\" in", values, "bankcardId");
            return (Criteria) this;
        }

        public Criteria andBankcardIdNotIn(List<String> values) {
            addCriterion("\"BANKCARD_ID\" not in", values, "bankcardId");
            return (Criteria) this;
        }

        public Criteria andBankcardIdBetween(String value1, String value2) {
            addCriterion("\"BANKCARD_ID\" between", value1, value2, "bankcardId");
            return (Criteria) this;
        }

        public Criteria andBankcardIdNotBetween(String value1, String value2) {
            addCriterion("\"BANKCARD_ID\" not between", value1, value2, "bankcardId");
            return (Criteria) this;
        }

        public Criteria andUserIdIsNull() {
            addCriterion("\"USER_ID\" is null");
            return (Criteria) this;
        }

        public Criteria andUserIdIsNotNull() {
            addCriterion("\"USER_ID\" is not null");
            return (Criteria) this;
        }

        public Criteria andUserIdEqualTo(String value) {
            addCriterion("\"USER_ID\" =", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdNotEqualTo(String value) {
            addCriterion("\"USER_ID\" <>", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdGreaterThan(String value) {
            addCriterion("\"USER_ID\" >", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdGreaterThanOrEqualTo(String value) {
            addCriterion("\"USER_ID\" >=", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdLessThan(String value) {
            addCriterion("\"USER_ID\" <", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdLessThanOrEqualTo(String value) {
            addCriterion("\"USER_ID\" <=", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdLike(String value) {
            addCriterion("\"USER_ID\" like", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdNotLike(String value) {
            addCriterion("\"USER_ID\" not like", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdIn(List<String> values) {
            addCriterion("\"USER_ID\" in", values, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdNotIn(List<String> values) {
            addCriterion("\"USER_ID\" not in", values, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdBetween(String value1, String value2) {
            addCriterion("\"USER_ID\" between", value1, value2, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdNotBetween(String value1, String value2) {
            addCriterion("\"USER_ID\" not between", value1, value2, "userId");
            return (Criteria) this;
        }

        public Criteria andBalanceIsNull() {
            addCriterion("\"BALANCE\" is null");
            return (Criteria) this;
        }

        public Criteria andBalanceIsNotNull() {
            addCriterion("\"BALANCE\" is not null");
            return (Criteria) this;
        }

        public Criteria andBalanceEqualTo(BigDecimal value) {
            addCriterion("\"BALANCE\" =", value, "balance");
            return (Criteria) this;
        }

        public Criteria andBalanceNotEqualTo(BigDecimal value) {
            addCriterion("\"BALANCE\" <>", value, "balance");
            return (Criteria) this;
        }

        public Criteria andBalanceGreaterThan(BigDecimal value) {
            addCriterion("\"BALANCE\" >", value, "balance");
            return (Criteria) this;
        }

        public Criteria andBalanceGreaterThanOrEqualTo(BigDecimal value) {
            addCriterion("\"BALANCE\" >=", value, "balance");
            return (Criteria) this;
        }

        public Criteria andBalanceLessThan(BigDecimal value) {
            addCriterion("\"BALANCE\" <", value, "balance");
            return (Criteria) this;
        }

        public Criteria andBalanceLessThanOrEqualTo(BigDecimal value) {
            addCriterion("\"BALANCE\" <=", value, "balance");
            return (Criteria) this;
        }

        public Criteria andBalanceIn(List<BigDecimal> values) {
            addCriterion("\"BALANCE\" in", values, "balance");
            return (Criteria) this;
        }

        public Criteria andBalanceNotIn(List<BigDecimal> values) {
            addCriterion("\"BALANCE\" not in", values, "balance");
            return (Criteria) this;
        }

        public Criteria andBalanceBetween(BigDecimal value1, BigDecimal value2) {
            addCriterion("\"BALANCE\" between", value1, value2, "balance");
            return (Criteria) this;
        }

        public Criteria andBalanceNotBetween(BigDecimal value1, BigDecimal value2) {
            addCriterion("\"BALANCE\" not between", value1, value2, "balance");
            return (Criteria) this;
        }

        public Criteria andDealTypeIsNull() {
            addCriterion("\"DEAL_TYPE\" is null");
            return (Criteria) this;
        }

        public Criteria andDealTypeIsNotNull() {
            addCriterion("\"DEAL_TYPE\" is not null");
            return (Criteria) this;
        }

        public Criteria andDealTypeEqualTo(String value) {
            addCriterion("\"DEAL_TYPE\" =", value, "dealType");
            return (Criteria) this;
        }

        public Criteria andDealTypeNotEqualTo(String value) {
            addCriterion("\"DEAL_TYPE\" <>", value, "dealType");
            return (Criteria) this;
        }

        public Criteria andDealTypeGreaterThan(String value) {
            addCriterion("\"DEAL_TYPE\" >", value, "dealType");
            return (Criteria) this;
        }

        public Criteria andDealTypeGreaterThanOrEqualTo(String value) {
            addCriterion("\"DEAL_TYPE\" >=", value, "dealType");
            return (Criteria) this;
        }

        public Criteria andDealTypeLessThan(String value) {
            addCriterion("\"DEAL_TYPE\" <", value, "dealType");
            return (Criteria) this;
        }

        public Criteria andDealTypeLessThanOrEqualTo(String value) {
            addCriterion("\"DEAL_TYPE\" <=", value, "dealType");
            return (Criteria) this;
        }

        public Criteria andDealTypeLike(String value) {
            addCriterion("\"DEAL_TYPE\" like", value, "dealType");
            return (Criteria) this;
        }

        public Criteria andDealTypeNotLike(String value) {
            addCriterion("\"DEAL_TYPE\" not like", value, "dealType");
            return (Criteria) this;
        }

        public Criteria andDealTypeIn(List<String> values) {
            addCriterion("\"DEAL_TYPE\" in", values, "dealType");
            return (Criteria) this;
        }

        public Criteria andDealTypeNotIn(List<String> values) {
            addCriterion("\"DEAL_TYPE\" not in", values, "dealType");
            return (Criteria) this;
        }

        public Criteria andDealTypeBetween(String value1, String value2) {
            addCriterion("\"DEAL_TYPE\" between", value1, value2, "dealType");
            return (Criteria) this;
        }

        public Criteria andDealTypeNotBetween(String value1, String value2) {
            addCriterion("\"DEAL_TYPE\" not between", value1, value2, "dealType");
            return (Criteria) this;
        }

        public Criteria andDealDateIsNull() {
            addCriterion("\"DEAL_DATE\" is null");
            return (Criteria) this;
        }

        public Criteria andDealDateIsNotNull() {
            addCriterion("\"DEAL_DATE\" is not null");
            return (Criteria) this;
        }

        public Criteria andDealDateEqualTo(Date value) {
            addCriterion("\"DEAL_DATE\" =", value, "dealDate");
            return (Criteria) this;
        }

        public Criteria andDealDateNotEqualTo(Date value) {
            addCriterion("\"DEAL_DATE\" <>", value, "dealDate");
            return (Criteria) this;
        }

        public Criteria andDealDateGreaterThan(Date value) {
            addCriterion("\"DEAL_DATE\" >", value, "dealDate");
            return (Criteria) this;
        }

        public Criteria andDealDateGreaterThanOrEqualTo(Date value) {
            addCriterion("\"DEAL_DATE\" >=", value, "dealDate");
            return (Criteria) this;
        }

        public Criteria andDealDateLessThan(Date value) {
            addCriterion("\"DEAL_DATE\" <", value, "dealDate");
            return (Criteria) this;
        }

        public Criteria andDealDateLessThanOrEqualTo(Date value) {
            addCriterion("\"DEAL_DATE\" <=", value, "dealDate");
            return (Criteria) this;
        }

        public Criteria andDealDateIn(List<Date> values) {
            addCriterion("\"DEAL_DATE\" in", values, "dealDate");
            return (Criteria) this;
        }

        public Criteria andDealDateNotIn(List<Date> values) {
            addCriterion("\"DEAL_DATE\" not in", values, "dealDate");
            return (Criteria) this;
        }

        public Criteria andDealDateBetween(Date value1, Date value2) {
            addCriterion("\"DEAL_DATE\" between", value1, value2, "dealDate");
            return (Criteria) this;
        }

        public Criteria andDealDateNotBetween(Date value1, Date value2) {
            addCriterion("\"DEAL_DATE\" not between", value1, value2, "dealDate");
            return (Criteria) this;
        }
    }

    public static class Criteria extends GeneratedCriteria {

        protected Criteria() {
            super();
        }
    }

    public static class Criterion {
        private String condition;

        private Object value;

        private Object secondValue;

        private boolean noValue;

        private boolean singleValue;

        private boolean betweenValue;

        private boolean listValue;

        private String typeHandler;

        public String getCondition() {
            return condition;
        }

        public Object getValue() {
            return value;
        }

        public Object getSecondValue() {
            return secondValue;
        }

        public boolean isNoValue() {
            return noValue;
        }

        public boolean isSingleValue() {
            return singleValue;
        }

        public boolean isBetweenValue() {
            return betweenValue;
        }

        public boolean isListValue() {
            return listValue;
        }

        public String getTypeHandler() {
            return typeHandler;
        }

        protected Criterion(String condition) {
            super();
            this.condition = condition;
            this.typeHandler = null;
            this.noValue = true;
        }

        protected Criterion(String condition, Object value, String typeHandler) {
            super();
            this.condition = condition;
            this.value = value;
            this.typeHandler = typeHandler;
            if (value instanceof List<?>) {
                this.listValue = true;
            } else {
                this.singleValue = true;
            }
        }

        protected Criterion(String condition, Object value) {
            this(condition, value, null);
        }

        protected Criterion(String condition, Object value, Object secondValue, String typeHandler) {
            super();
            this.condition = condition;
            this.value = value;
            this.secondValue = secondValue;
            this.typeHandler = typeHandler;
            this.betweenValue = true;
        }

        protected Criterion(String condition, Object value, Object secondValue) {
            this(condition, value, secondValue, null);
        }
    }
}