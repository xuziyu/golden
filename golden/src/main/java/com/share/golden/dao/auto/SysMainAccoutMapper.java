package com.share.golden.dao.auto;

import com.share.golden.entity.SysMainAccout;
import com.share.golden.entity.SysMainAccoutExample;
import java.util.List;
import org.apache.ibatis.annotations.Param;

public interface SysMainAccoutMapper {
    long countByExample(SysMainAccoutExample example);

    int deleteByExample(SysMainAccoutExample example);

    int deleteByPrimaryKey(String dealId);

    int insert(SysMainAccout record);

    int insertSelective(SysMainAccout record);

    List<SysMainAccout> selectByExample(SysMainAccoutExample example);

    SysMainAccout selectByPrimaryKey(String dealId);

    int updateByExampleSelective(@Param("record") SysMainAccout record, @Param("example") SysMainAccoutExample example);

    int updateByExample(@Param("record") SysMainAccout record, @Param("example") SysMainAccoutExample example);

    int updateByPrimaryKeySelective(SysMainAccout record);

    int updateByPrimaryKey(SysMainAccout record);
}