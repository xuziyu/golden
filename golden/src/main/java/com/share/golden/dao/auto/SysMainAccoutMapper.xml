<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE mapper PUBLIC "-//mybatis.org//DTD Mapper 3.0//EN" "http://mybatis.org/dtd/mybatis-3-mapper.dtd">
<mapper namespace="com.share.golden.dao.auto.SysMainAccoutMapper">
  <resultMap id="BaseResultMap" type="com.share.golden.entity.SysMainAccout">
    <id column="DEAL_ID" jdbcType="VARCHAR" property="dealId" />
    <result column="BANKCARD_ID" jdbcType="VARCHAR" property="bankcardId" />
    <result column="USER_ID" jdbcType="VARCHAR" property="userId" />
    <result column="BALANCE" jdbcType="DECIMAL" property="balance" />
    <result column="DEAL_TYPE" jdbcType="VARCHAR" property="dealType" />
    <result column="DEAL_DATE" jdbcType="TIMESTAMP" property="dealDate" />
  </resultMap>
  <sql id="Example_Where_Clause">
    <where>
      <foreach collection="oredCriteria" item="criteria" separator="or">
        <if test="criteria.valid">
          <trim prefix="(" prefixOverrides="and" suffix=")">
            <foreach collection="criteria.criteria" item="criterion">
              <choose>
                <when test="criterion.noValue">
                  and ${criterion.condition}
                </when>
                <when test="criterion.singleValue">
                  and ${criterion.condition} #{criterion.value}
                </when>
                <when test="criterion.betweenValue">
                  and ${criterion.condition} #{criterion.value} and #{criterion.secondValue}
                </when>
                <when test="criterion.listValue">
                  and ${criterion.condition}
                  <foreach close=")" collection="criterion.value" item="listItem" open="(" separator=",">
                    #{listItem}
                  </foreach>
                </when>
              </choose>
            </foreach>
          </trim>
        </if>
      </foreach>
    </where>
  </sql>
  <sql id="Update_By_Example_Where_Clause">
    <where>
      <foreach collection="example.oredCriteria" item="criteria" separator="or">
        <if test="criteria.valid">
          <trim prefix="(" prefixOverrides="and" suffix=")">
            <foreach collection="criteria.criteria" item="criterion">
              <choose>
                <when test="criterion.noValue">
                  and ${criterion.condition}
                </when>
                <when test="criterion.singleValue">
                  and ${criterion.condition} #{criterion.value}
                </when>
                <when test="criterion.betweenValue">
                  and ${criterion.condition} #{criterion.value} and #{criterion.secondValue}
                </when>
                <when test="criterion.listValue">
                  and ${criterion.condition}
                  <foreach close=")" collection="criterion.value" item="listItem" open="(" separator=",">
                    #{listItem}
                  </foreach>
                </when>
              </choose>
            </foreach>
          </trim>
        </if>
      </foreach>
    </where>
  </sql>
  <sql id="Base_Column_List">
    "DEAL_ID", "BANKCARD_ID", "USER_ID", "BALANCE", "DEAL_TYPE", "DEAL_DATE"
  </sql>
  <select id="selectByExample" parameterType="com.share.golden.entity.SysMainAccoutExample" resultMap="BaseResultMap">
    select
    <if test="distinct">
      distinct
    </if>
    <include refid="Base_Column_List" />
    from SYS_MAIN_ACCOUT
    <if test="_parameter != null">
      <include refid="Example_Where_Clause" />
    </if>
    <if test="orderByClause != null">
      order by ${orderByClause}
    </if>
  </select>
  <select id="selectByPrimaryKey" parameterType="java.lang.String" resultMap="BaseResultMap">
    select 
    <include refid="Base_Column_List" />
    from SYS_MAIN_ACCOUT
    where "DEAL_ID" = #{dealId,jdbcType=VARCHAR}
  </select>
  <delete id="deleteByPrimaryKey" parameterType="java.lang.String">
    delete from SYS_MAIN_ACCOUT
    where "DEAL_ID" = #{dealId,jdbcType=VARCHAR}
  </delete>
  <delete id="deleteByExample" parameterType="com.share.golden.entity.SysMainAccoutExample">
    delete from SYS_MAIN_ACCOUT
    <if test="_parameter != null">
      <include refid="Example_Where_Clause" />
    </if>
  </delete>
  <insert id="insert" parameterType="com.share.golden.entity.SysMainAccout">
    insert into SYS_MAIN_ACCOUT ("DEAL_ID", "BANKCARD_ID", "USER_ID", 
      "BALANCE", "DEAL_TYPE", "DEAL_DATE"
      )
    values (#{dealId,jdbcType=VARCHAR}, #{bankcardId,jdbcType=VARCHAR}, #{userId,jdbcType=VARCHAR}, 
      #{balance,jdbcType=DECIMAL}, #{dealType,jdbcType=VARCHAR}, #{dealDate,jdbcType=TIMESTAMP}
      )
  </insert>
  <insert id="insertSelective" parameterType="com.share.golden.entity.SysMainAccout">
    insert into SYS_MAIN_ACCOUT
    <trim prefix="(" suffix=")" suffixOverrides=",">
      <if test="dealId != null">
        "DEAL_ID",
      </if>
      <if test="bankcardId != null">
        "BANKCARD_ID",
      </if>
      <if test="userId != null">
        "USER_ID",
      </if>
      <if test="balance != null">
        "BALANCE",
      </if>
      <if test="dealType != null">
        "DEAL_TYPE",
      </if>
      <if test="dealDate != null">
        "DEAL_DATE",
      </if>
    </trim>
    <trim prefix="values (" suffix=")" suffixOverrides=",">
      <if test="dealId != null">
        #{dealId,jdbcType=VARCHAR},
      </if>
      <if test="bankcardId != null">
        #{bankcardId,jdbcType=VARCHAR},
      </if>
      <if test="userId != null">
        #{userId,jdbcType=VARCHAR},
      </if>
      <if test="balance != null">
        #{balance,jdbcType=DECIMAL},
      </if>
      <if test="dealType != null">
        #{dealType,jdbcType=VARCHAR},
      </if>
      <if test="dealDate != null">
        #{dealDate,jdbcType=TIMESTAMP},
      </if>
    </trim>
  </insert>
  <select id="countByExample" parameterType="com.share.golden.entity.SysMainAccoutExample" resultType="java.lang.Long">
    select count(*) from SYS_MAIN_ACCOUT
    <if test="_parameter != null">
      <include refid="Example_Where_Clause" />
    </if>
  </select>
  <update id="updateByExampleSelective" parameterType="map">
    update SYS_MAIN_ACCOUT
    <set>
      <if test="record.dealId != null">
        "DEAL_ID" = #{record.dealId,jdbcType=VARCHAR},
      </if>
      <if test="record.bankcardId != null">
        "BANKCARD_ID" = #{record.bankcardId,jdbcType=VARCHAR},
      </if>
      <if test="record.userId != null">
        "USER_ID" = #{record.userId,jdbcType=VARCHAR},
      </if>
      <if test="record.balance != null">
        "BALANCE" = #{record.balance,jdbcType=DECIMAL},
      </if>
      <if test="record.dealType != null">
        "DEAL_TYPE" = #{record.dealType,jdbcType=VARCHAR},
      </if>
      <if test="record.dealDate != null">
        "DEAL_DATE" = #{record.dealDate,jdbcType=TIMESTAMP},
      </if>
    </set>
    <if test="_parameter != null">
      <include refid="Update_By_Example_Where_Clause" />
    </if>
  </update>
  <update id="updateByExample" parameterType="map">
    update SYS_MAIN_ACCOUT
    set "DEAL_ID" = #{record.dealId,jdbcType=VARCHAR},
      "BANKCARD_ID" = #{record.bankcardId,jdbcType=VARCHAR},
      "USER_ID" = #{record.userId,jdbcType=VARCHAR},
      "BALANCE" = #{record.balance,jdbcType=DECIMAL},
      "DEAL_TYPE" = #{record.dealType,jdbcType=VARCHAR},
      "DEAL_DATE" = #{record.dealDate,jdbcType=TIMESTAMP}
    <if test="_parameter != null">
      <include refid="Update_By_Example_Where_Clause" />
    </if>
  </update>
  <update id="updateByPrimaryKeySelective" parameterType="com.share.golden.entity.SysMainAccout">
    update SYS_MAIN_ACCOUT
    <set>
      <if test="bankcardId != null">
        "BANKCARD_ID" = #{bankcardId,jdbcType=VARCHAR},
      </if>
      <if test="userId != null">
        "USER_ID" = #{userId,jdbcType=VARCHAR},
      </if>
      <if test="balance != null">
        "BALANCE" = #{balance,jdbcType=DECIMAL},
      </if>
      <if test="dealType != null">
        "DEAL_TYPE" = #{dealType,jdbcType=VARCHAR},
      </if>
      <if test="dealDate != null">
        "DEAL_DATE" = #{dealDate,jdbcType=TIMESTAMP},
      </if>
    </set>
    where "DEAL_ID" = #{dealId,jdbcType=VARCHAR}
  </update>
  <update id="updateByPrimaryKey" parameterType="com.share.golden.entity.SysMainAccout">
    update SYS_MAIN_ACCOUT
    set "BANKCARD_ID" = #{bankcardId,jdbcType=VARCHAR},
      "USER_ID" = #{userId,jdbcType=VARCHAR},
      "BALANCE" = #{balance,jdbcType=DECIMAL},
      "DEAL_TYPE" = #{dealType,jdbcType=VARCHAR},
      "DEAL_DATE" = #{dealDate,jdbcType=TIMESTAMP}
    where "DEAL_ID" = #{dealId,jdbcType=VARCHAR}
  </update>
</mapper>